chef/cookbooks/neutron/recipes/common_agent.rb

```diff
@@ -306,14 +231,204 @@
   end
 end
 
-include_recipe "neutron::common_config"
+neutron_network_ha = node.roles.include?("neutron-network") && neutron[:neutron][:ha][:network][:enabled]
+
+# ML2 configuration: L2 agent and L3 agent
+if neutron[:neutron][:networking_plugin] == "ml2"
+  include_recipe "neutron::common_config"
+
+  ml2_mech_drivers = neutron[:neutron][:ml2_mechanism_drivers]
+  ml2_type_drivers = neutron[:neutron][:ml2_type_drivers]
 
-neutron_l3_ha = node.roles.include?("neutron-l3") && node[:neutron][:ha][:l3][:enabled]
+  case
+  when ml2_mech_drivers.include?("openvswitch")
+    neutron_agent = node[:neutron][:platform][:ovs_agent_name]
+    neutron_agent_pkg = node[:neutron][:platform][:ovs_agent_pkg]
+    agent_config_path = "/etc/neutron/plugins/openvswitch/ovs_neutron_plugin.ini"
+    interface_driver = "neutron.agent.linux.interface.OVSInterfaceDriver"
+    external_network_bridge = "br-public"
+  when ml2_mech_drivers.include?("linuxbridge")
+    neutron_agent = node[:neutron][:platform][:lb_agent_name]
+    neutron_agent_pkg = node[:neutron][:platform][:lb_agent_pkg]
+    agent_config_path = "/etc/neutron/plugins/linuxbridge/linuxbridge_conf.ini"
+    interface_driver = "neutron.agent.linux.interface.BridgeInterfaceDriver"
+    external_network_bridge = ""
+  end
 
-service neutron_agent do
+  # L2 agent
+  unless neutron[:neutron][:use_gitrepo]
+    package neutron_agent_pkg
```

Comment:
I think that you are moving this to the low part of the file, but there is a service that I think is related with this package, that is addressed before: `service_name "openstack-neutron-ovs-cleanup" if %w(suse).include?(node.platform)`